com.github.dictation_toolbox.Caster.yaml

```diff
@@ -0,0 +1,81 @@
+app-id: com.github.dictation_toolbox.Caster
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+command: python3 -m dragonfly load _caster.py --engine kaldi --no-recobs-messages
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --device=all
+
+build-options:
+    append-path: /usr/lib/sdk/rust-stable/bin
+
+cleanup:
+  - /lib/cmake
+  - /lib/pkgconfig
+  - /lib/wx
+  - /lib64
+  - /share/aclocal
+  - /share/bakefile
+  - '*.la'
+  - '*.a'
+
+modules:
+
+  - shared-modules/glu/glu-9.json
+  - shared-modules/glew/glew.json
+
+  - name: xdotool
+    no-autogen: true
+    make-args:
+      - WITHOUT_RPATH_FIX=1
+    make-install-args:
+      - PREFIX=${FLATPAK_DEST}
+    sources:
+      - type: archive
+        url: https://github.com/jordansissel/xdotool/releases/download/v3.20211022.1/xdotool-3.20211022.1.tar.gz
+        sha256: 96f0facfde6d78eacad35b91b0f46fecd0b35e474c03e00e30da3fdd345f9ada
+  - name: portaudio
+    buildsystem: autotools
+    sources:
+      - type: archive
+        url: https://github.com/PortAudio/portaudio/archive/refs/tags/v19.7.0.tar.gz
+        sha256: 5af29ba58bbdbb7bbcefaaecc77ec8fc413f0db6f4c4e286c40c3e1b83174fa0
+
+  - name: wxWidgets
+    cleanup:
+      - /bin
+      - /include
+    config-opts:
+      - --with-opengl
+      - --disable-glcanvasegl
+    sources:
+      - type: git
+        commit: 085a136dcb11aca5b1102193f006f8056d5f0876
+        tag: v3.2.4
+        url: https://github.com/wxWidgets/wxWidgets
+        x-checker-data:
+          is-important: true
+          type: git
+          tag-pattern: ^v([\d\.]+)$
+
+  - python3-wxPython.yml
+
+  - caster.yml
+
+  - name: caster
+    buildsystem: simple
+    build-commands:
+      - pip3 install --no-index --find-links="file://${PWD}" -r requirements-mac-linux.txt
+      - unzip kaldi_model_daanzu_20211030-mediumlm.zip -d ~/Caster/kaldi_model
```

Comment:
`type: archive` should be enough to extract it

Reply:
> `type: archive` should be enough to extract it I think

is this in reference to unzip? I need to have the model unzipped to the project dir, as this says:

https://github.com/dictation-toolbox/Caster/issues/849#issuecomment-725044490

Reply:
These are run at build time, how will you access the user's home directory? I don't know how it is supposed to be used, but you can move it to somewhere under `/app/kaldi_model`

Reply:
> Where does the app expect it to be preset?

According to this: https://github.com/dictation-toolbox/Caster/issues/849#issuecomment-725044490

> 4. Set up Kaldi Model

>    Download your preferred Kaldi model at [kaldi-active-grammar/releases](https://github.com/daanzu/kaldi-active-grammar/releases)

>    Extract kaldi_model_< Model Type >.zip to \home\Documents\Caster

Reply:
That won't work, how is it supposed to be used by the app? I see a bunch of .fst files in there that are ~500 MB.

Either the app can download and extract it at `~/.var/app/<app-id>/data` as part of its own data at runtime, or you can place them in somewhere under `/app` or keep it out of the flatpak and users will have downloaded this part.

Reply:
Thanks for this feedback. I have linked the Caster maintainers to this PR. I am awaiting their feedback on this issue and the program executing issue.